sklearn/decomposition/_nmf.py

```diff
@@ -222,6 +222,12 @@ def _check_string_param(solver, regularization, beta_loss, init):
             'Invalid beta_loss parameter: solver %r does not handle beta_loss'
             ' = %r' % (solver, beta_loss))
 
+    if batch_size is not None:
+        if beta_loss in (2, 'frobenius') or solver == 'cd':
+            raise ValueError("Invalid beta_loss parameter 'frobenius' "
+                             "or invalid solver 'cd' not supported "
+                             "when batch_size is not None.")
```

Comment:
Looking at the refs, it's really not obvious to me that other beta divergences don't work. Let's make some tests with general beta divergences before deciding when to raise an error.

Reply:
```suggestion
```